packages/gitbook/src/components/DocumentView/Table/utils.ts

```diff
@@ -12,6 +17,23 @@ export function getRecordValue<T extends number | string | boolean | string[] |
     return record.values[definitionId];
 }
 
+/**
+ * Get the covers for a record card.
+ * Returns both the light and dark covers.
+ * The light cover is a string or a content ref (image or files column type).
+ * The dark cover is a content ref (image column type).
+ */
+export function getRecordCardCovers(record: DocumentTableRecord, view: DocumentTableViewCards) {
+    return {
+        light: view.coverDefinition
```

Comment:
```suggestion
        light: (() => {
          if (!view.coverDefinition) {
            return null;
          }
          
          const value = getRecordValue(record, view.coverDefinition) as ContentRefFile | ContentRefURL | string[];
          if(Array.isArray(value) && typeof value[0] === 'string') {
            return { kind: 'file', file: value[0] };
          }
          
          return value;
        })()
```

Do all your legacy content ref checking here. Keep the return type of this util really clean - just return content refs.